sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClientIntegrationTest.java

```diff
@@ -83,6 +89,284 @@ protected void afterTest() {
+     * Verifies that we can complete a message with lock token only with a transaction and rollback.
+     */
+    @MethodSource("messagingEntityProvider")
+    @ParameterizedTest
+    void transactionWithLockTokenTest(MessagingEntityType entityType) {
+
+        // Arrange
+        boolean isSessionEnabled = false;
+        setSenderAndReceiver(entityType, isSessionEnabled);
+        ServiceBusReceiverAsyncClient receiverNonConnectionSharing = getReceiverBuilder(false, entityType,
+            Function.identity()).buildAsyncClient();
+
+        final String messageId = UUID.randomUUID().toString();
+        final ServiceBusMessage message = getMessage(messageId, isSessionEnabled);
+
+        sendMessage(message).block(TIMEOUT);
+
+        // Assert & Act
+        AtomicReference<ServiceBusTransactionContext> transaction = new AtomicReference<>();
+
+        // create a transaction.
+        StepVerifier.create(receiverNonConnectionSharing.createTransaction())
+            .assertNext(txn -> {
+                transaction.set(txn);
+                assertNotNull(transaction);
+            })
+            .verifyComplete();
+
+        AtomicReference<MessageLockToken> messageLockToken = new AtomicReference<>();
+
+        // receive a message and get lock token.
+        StepVerifier.create(receiver.receive().next()
+            .map(messageContext -> {
+                ServiceBusReceivedMessage received =  messageContext.getMessage();
+                messageLockToken.set(MessageLockToken.fromString(received.getLockToken()));
+                return messageContext;
+            }))
+            .assertNext(receivedMessage -> assertMessageEquals(receivedMessage, messageId, isSessionEnabled))
+            .verifyComplete();
+
+        // complete the message using lock token only using a receiver which represent a different machine
+        StepVerifier.create(receiverNonConnectionSharing.complete(messageLockToken.get(), transaction.get()))
+            .verifyComplete();
+
+        // rollback the transaction.
+        StepVerifier.create(receiverNonConnectionSharing.rollbackTransaction(transaction.get()))
+            .verifyComplete();
+
+        // read the message back, since it was rolled-back previously.
+        final ServiceBusReceivedMessageContext receivedContext = receiveAndDeleteReceiver.receive().next().block(TIMEOUT);
+        assertMessageEquals(receivedContext, messageId, isSessionEnabled);
+
+        messagesPending.decrementAndGet();
+    }
+
+    /**
+     * Verifies that we can do following using shared connection.
+     * 1. create transaction
+     * 2. send message  with transactionContext
+     * 3. receive and settle with transactionContext.
+     * 4. commit Rollback this transaction.
+     */
+    @MethodSource("messagingEntityTxnWithSessions")
+    @ParameterizedTest
+    void transactionSendReceiveAndSettle(MessagingEntityType entityType, boolean isSessionEnabled,
```

Comment:
It's not valuable to test every combination. It makes this exponentially longer to run. Choose a couple.